File: test-websocket-client.py
#!/usr/bin/env python3
"""
WebSocket Test Client - Tests Socket.IO connection
"""

import socketio
import sys
import time

CONVERSATION_ID = sys.argv[1] if len(sys.argv) > 1 else '42327374-aef7-4b04-81b9-0e54ce42a44e'

print(f"🔌 Connecting to WebSocket server...")
print(f"   URL: http://localhost:3000")
print(f"   Conversation ID: {CONVERSATION_ID}\n")

sio = socketio.Client()

@sio.event
def connect():
    print(f"✅ Connected to server")
    print(f"   Socket ID: {sio.sid}\n")
    
    # Join conversation room
    print(f"📬 Joining conversation: {CONVERSATION_ID}")
    sio.emit('join:conversation', {
        'conversationId': CONVERSATION_ID,
        'username': 'python-test-client'
    })

@sio.event
def joined(data):
    print("✅ Successfully joined conversation")
    print(f"   Data: {data}")
    print("\n👂 Listening for events...\n")

@sio.event
def disconnect():
    print("❌ Disconnected from server")

@sio.on('agent:spawned')
def on_agent_spawned(data):
    print("🤖 Agent Spawned:")
    print(f"   {data}")

@sio.on('agent:status')
def on_agent_status(data):
    print("📊 Agent Status:")
    print(f"   {data}")

@sio.on('agent:message')
def on_agent_message(data):
    print("💬 Agent Message:")
    print(f"   {data}")

@sio.on('task:created')
def on_task_created(data):
    print("📝 Task Created:")
    print(f"   {data}")

@sio.on('task:updated')
def on_task_updated(data):
    print("🔄 Task Updated:")
    print(f"   {data}")

@sio.on('query:progress')
def on_query_progress(data):
    print("⏳ Query Progress:")
    print(f"   {data}")

@sio.on('query:completed')
def on_query_completed(data):
    print("✅ Query Completed:")
    print(f"   {data}")

@sio.on('error')
def on_error(data):
    print("❌ Error:")
    print(f"   {data}")

try:
    sio.connect('http://localhost:3000')
    
    print("Press Ctrl+C to exit\n")
    
    # Keep running
    while True:
        time.sleep(1)
        
except KeyboardInterrupt:
    print("\n\n👋 Disconnecting...")
    sio.disconnect()
except Exception as e:
    print(f"❌ Error: {e}")
    sys.exit(1)

File: test-websocket-client.mjs
#!/usr/bin/env node

/**
 * WebSocket Test Client - Tests Socket.IO connection and event handling
 */

import { io } from 'socket.io-client';

const SOCKET_URL = 'http://localhost:3000';
const CONVERSATION_ID = process.argv[2] || '0717ba8b-83a2-41c5-a9eb-e5bf283e0aef';

console.log('🔌 Connecting to WebSocket server...');
console.log(`   URL: ${SOCKET_URL}`);
console.log(`   Conversation ID: ${CONVERSATION_ID}\n`);

const socket = io(SOCKET_URL, {
  transports: ['websocket', 'polling'],
});

// Connection events
socket.on('connect', () => {
  console.log('✅ Connected to server');
  console.log(`   Socket ID: ${socket.id}\n`);

  // Join a conversation room
  console.log(`📬 Joining conversation: ${CONVERSATION_ID}`);
  socket.emit('join:conversation', {
    conversationId: CONVERSATION_ID,
    username: 'test-client',
  });
});

socket.on('joined', (data) => {
  console.log('✅ Successfully joined conversation');
  console.log('   Data:', JSON.stringify(data, null, 2));
  console.log('\n👂 Listening for events...\n');
});

socket.on('disconnect', () => {
  console.log('❌ Disconnected from server');
});

socket.on('connect_error', (error) => {
  console.error('❌ Connection error:', error.message);
  process.exit(1);
});

// Agent events
socket.on('agent:spawned', (data) => {
  console.log('🤖 Agent Spawned:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

socket.on('agent:status', (data) => {
  console.log('📊 Agent Status:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

socket.on('agent:message', (data) => {
  console.log('💬 Agent Message:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

// Task events
socket.on('task:created', (data) => {
  console.log('📝 Task Created:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

socket.on('task:updated', (data) => {
  console.log('🔄 Task Updated:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

// Query events
socket.on('query:progress', (data) => {
  console.log('⏳ Query Progress:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

socket.on('query:completed', (data) => {
  console.log('✅ Query Completed:');
  console.log('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

// Error events
socket.on('error', (data) => {
  console.error('❌ Error:');
  console.error('   ' + JSON.stringify(data, null, 2).replace(/\n/g, '\n   '));
});

// Keep the process running
console.log('Press Ctrl+C to exit\n');
process.on('SIGINT', () => {
  console.log('\n\n👋 Disconnecting...');
  socket.disconnect();
  process.exit(0);
});
